src/core/stream/stream_loader.ts

```diff
@@ -194,6 +220,27 @@ export default function StreamLoader({
     const stallingManager$ = StallingManager(mediaElement, clock$)
       .pipe(map(EVENTS.stalled));
 
+    // Creates an observable which will refresh the manifest after
+    // the update period has elapsed.
+    const updateManifest$ = refreshMinimumUpdatePeriod$.pipe(
+      mergeMap((lastUpdatePeriod) => {
+        return observableTimer(lastUpdatePeriod * 1000).pipe(
```

Comment:
To think of it the `performance.now() - requestTime` difference might even be calculated here... To see if that makes sense and if it is readable